### PR TITLE
x86: add some common Kconfig symbols

### PR DESCRIPTION
**Pull request by graysky2**
* Build a tickless kernel which can benefit in lower power consumption.
* Enable framework from which sybsystems can access energy models.
* Expose pressure stall info.
* Expose RAPL sysfs interface.
* Enable thermal framework's netlink interface which has been recommended by upstream.
* Use Hardware Feedback Interface for thermals.

Build system: x86/64
Build-tested: x86/64
Run-tested: x86/64

### CODE REVIEW

**Comment by graysky2** [comment]
I am not sure if `CONFIG_AMD_SFH_HID=y` should be made a module or not...

**Comment by graysky2** [comment]
I was able to build this with external toolchain without errors.  Something in the CI?
```
% cat diffconfig
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_64=y
CONFIG_TARGET_x86_64_DEVICE_generic=y
CONFIG_DEVEL=y
CONFIG_BUILD_NLS=y
CONFIG_EXTERNAL_TOOLCHAIN=y
CONFIG_EXTERNAL_TOOLCHAIN_LIBC_USE_MUSL=y
CONFIG_LIBC_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/ld-musl-x86_64.so.1 ./x86_64-openwrt-linux-musl/lib/libc.so"
CONFIG_LIBC_ROOT_DIR="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBGCC_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/libgcc_s.so ./x86_64-openwrt-linux-musl/lib/libgcc_s.so.1"
CONFIG_LIBGCC_ROOT_DIR="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBPTHREAD_FILE_SPEC="./lib/libpthread{-*.so,.so.*}"
CONFIG_LIBPTHREAD_ROOT_DIR="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBRT_FILE_SPEC="./lib/librt{-*.so,.so.*}"
CONFIG_LIBRT_ROOT_DIR="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBSTDCPP_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/libstdc++.so ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6 ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6.0.30 ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6.0.30-gdb.py"
CONFIG_LIBSTDCPP_ROOT_DIR="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_PACKAGE_libstdcpp=y
CONFIG_TARGET_NAME="x86_64-openwrt-linux-musl"
CONFIG_TOOLCHAIN_BIN_PATH="./usr/bin ./bin"
CONFIG_TOOLCHAIN_INC_PATH="./usr/include ./include/fortify ./include"
CONFIG_TOOLCHAIN_LIBC="musl"
CONFIG_TOOLCHAIN_LIB_PATH="./usr/lib ./lib"
CONFIG_TOOLCHAIN_PREFIX="x86_64-openwrt-linux-musl-"
CONFIG_TOOLCHAIN_ROOT="/scratch/ok/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_USE_EXTERNAL_LIBC=y
```

**Comment by hauke** [comment]
It fails like this:
````
====== Make errors from logs/target/linux/compile.txt ======
make[4]: Leaving directory '/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-5.15.120'
.config:1225:warning: override: CPU_FREQ_DEFAULT_GOV_SCHEDUTIL changes choice state
.config:3947:warning: override: M686 changes choice state
.config:7382:warning: override: SPARSEMEM_MANUAL changes choice state
arch/x86/kernel/head_64.o: warning: objtool: .text+0x5: unreachable instruction
arch/x86/kernel/smp.o: warning: objtool: sysvec_reboot()+0x51: unreachable instruction
mm/memory.c:91:2: error: #warning Unfortunate NUMA and NUMA Balancing config, growing page-frame for last_cpupid. [-Werror=cpp]
   91 | #warning Unfortunate NUMA and NUMA Balancing config, growing page-frame for last_cpupid.
      |  ^~~~~~~
cc1: all warnings being treated as errors
make[5]: *** [scripts/Makefile.build:289: mm/memory.o] Error 1
make[4]: *** [Makefile:1906: mm] Error 2
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [Makefile:25: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-5.15.120/.modules] Error 2
make[2]: *** [Makefile:11: compile] Error 2
time: target/linux/compile#438.34#53.29#262.25
Error: Process completed with exit code 2.
````

It uses this configuration:
````
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_64=y
CONFIG_TARGET_x86_64_DEVICE_generic=y
CONFIG_ALL_KMODS=y
CONFIG_DEVEL=y
CONFIG_AUTOREMOVE=y
CONFIG_BUILD_NLS=y
CONFIG_CCACHE=y
CONFIG_EXTERNAL_TOOLCHAIN=y
CONFIG_EXTERNAL_TOOLCHAIN_LIBC_USE_MUSL=y
CONFIG_KERNEL_WERROR=y
CONFIG_LIBC_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/ld-musl-x86_64.so.1 ./x86_64-openwrt-linux-musl/lib/libc.so"
CONFIG_LIBC_ROOT_DIR="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBGCC_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/libgcc_s.so ./x86_64-openwrt-linux-musl/lib/libgcc_s.so.1"
CONFIG_LIBGCC_ROOT_DIR="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBPTHREAD_FILE_SPEC="./lib/libpthread{-*.so,.so.*}"
CONFIG_LIBPTHREAD_ROOT_DIR="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBRT_FILE_SPEC="./lib/librt{-*.so,.so.*}"
CONFIG_LIBRT_ROOT_DIR="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_LIBSTDCPP_FILE_SPEC="./x86_64-openwrt-linux-musl/lib/libstdc++.so ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6 ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6.0.30 ./x86_64-openwrt-linux-musl/lib/libstdc++.so.6.0.30-gdb.py"
CONFIG_LIBSTDCPP_ROOT_DIR="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_MOLD=y
CONFIG_PACKAGE_libstdcpp=y
CONFIG_TARGET_NAME="x86_64-openwrt-linux-musl"
CONFIG_TOOLCHAIN_BIN_PATH="./usr/bin ./bin"
CONFIG_TOOLCHAIN_INC_PATH="./usr/include ./include/fortify ./include"
CONFIG_TOOLCHAIN_LIBC="musl"
CONFIG_TOOLCHAIN_LIB_PATH="./usr/lib ./lib"
CONFIG_TOOLCHAIN_PREFIX="x86_64-openwrt-linux-musl-"
CONFIG_TOOLCHAIN_ROOT="/external-toolchain/openwrt-toolchain-x86-64_gcc-12.3.0_musl.Linux-x86_64/toolchain-x86_64_gcc-12.3.0_musl"
CONFIG_USE_EXTERNAL_LIBC=y
CONFIG_MOLD=y
# CONFIG_USE_MOLD is not set
````

Probably activating all kernel modules causes this problem.

**Comment by graysky2** [comment]
Thanks @hauke - I am not sure what I need to do in order for the CI external toolchain build to work properly.  Can you advise?

**Comment by trippleflux** [comment]
Some typos : 
```
CONFIG_NUMA_EMU=n
CONFIG_CONTEXT_TRACKING_USER_FORCE=n
CONFIG_RCU_NOCB_CPU_DEFAULT_ALL=n
```

Should be : 
```
# CONFIG_NUMA_EMU is not set
# CONFIG_CONTEXT_TRACKING_USER_FORCE is not set
# CONFIG_RCU_NOCB_CPU_DEFAULT_ALL is not set
```

**Comment by hauke** [comment]
Probably you do not need an external toolchain. This will probably already show the problem_
````
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_64=y
CONFIG_TARGET_x86_64_DEVICE_generic=y
CONFIG_ALL_KMODS=y
````

**Comment by graysky2** [comment]
@trippleflux - thank you, I corrected them.

**Comment by graysky2** [comment]
@hauke - you are correct... getting an error in mac80211:
```
...
make[3]: Entering directory '/scratch/union/package/kernel/mac80211'
troff:vswitchd/ovs-vswitchd.8:1290: warning: cannot select font 'CW'
broadcom.mk:378: warning: overriding recipe for target '/scratch/union/dl/broadcom-wl-5.100.138.tar.bz2'
broadcom.mk:95: warning: ignoring old recipe for target '/scratch/union/dl/broadcom-wl-5.100.138.tar.bz2'
make[6]: *** [Makefile:5684: manpage-check] Error 1
make[6]: Leaving directory '/scratch/union/build_dir/target-x86_64_musl/linux-x86_64/openvswitch-2.17.0'
make[5]: *** [Makefile:4431: all-recursive] Error 1
make[5]: Leaving directory '/scratch/union/build_dir/target-x86_64_musl/linux-x86_64/openvswitch-2.17.0'
make[4]: *** [Makefile:2333: all] Error 2
make[4]: Leaving directory '/scratch/union/build_dir/target-x86_64_musl/linux-x86_64/openvswitch-2.17.0'
make[3]: *** [Makefile:280: /scratch/union/build_dir/target-x86_64_musl/linux-x86_64/openvswitch-2.17.0/.built] Error 2
make[3]: Leaving directory '/scratch/union/feeds/packages/net/openvswitch'
time: package/feeds/packages/openvswitch/compile#0.79#0.14#0.86
    ERROR: package/feeds/packages/openvswitch failed to build.
make[2]: *** [package/Makefile:120: package/feeds/packages/openvswitch/compile] Error 1
...
```

**Comment by trippleflux** [comment]
@graysky2 

Have you tried : 
```
make package/openvswitch/{clean,compile} -j1 V=s
```

To pin point the real causes from openvswitch package?

**Comment by trippleflux** [comment]
Some notes that I have found regarding NUMA : 

**_Regarding NUMA in general from gentoo's users_** [discussion](https://groups.google.com/g/linux.gentoo.user/c/aGGO2p1ZCNw).

**_[Optimal Configuration of Memory System](https://support.huawei.com/enterprise/en/doc/EDOC1000117211/1395f4a4/optimal-configuration-of-memory-system)_**

_**Automatic NUMA Balancing**_
Automatic NUMA balancing is a new feature introduced from **_Linux 3.8 Kernel._** It can automatically optimize the application running in the NUMA system. It is enabled in the RHEL 7 system by default.

For applications such as Oracle, NUMA optimization has been performed. It is advised not to enable this function, which is called by the applications based on optimal mode. Fort some scenarios requiring manual optimization by numactl, it is also advised not to enable this function.

Disabling method:

```
echo 0 > /proc/sys/kernel/numa_balancing

or

sysctl -w kernel.numa_balancing=0
```
It is valid permanently and written into the sysctl.conf file. It is validated using sysctl –p.

> BUT Documentation above based on finding and benchmark from really old kernel version 3.8!
> Newer RHEL docs mentioned otherwise, in [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_tuning_and_optimization_guide/sect-virtualization_tuning_optimization_guide-numa-auto_numa_balancing).

An interesting webui for some distro default kernel config for certain kernel config var.
[CONFIG_NUMA_BALANCING](https://hlandau.github.io/kconfigreport/option/CONFIG_NUMA_BALANCING.xhtml)
[CONFIG_NUMA_BALANCING_DEFAULT_ENABLED](https://hlandau.github.io/kconfigreport/option/CONFIG_NUMA_BALANCING_DEFAULT_ENABLED.xhtml)

Online browsable interesting sources / headers docs regarding [NUMA balancing](https://elixir.bootlin.com/linux/v5.15.120/source/include/linux/page-flags-layout.h#L36).

I have a mixed feelings regarding enabling NUMA in general. Perhaps we needs more testing?.
Who wanted to run OpenWrt on dual socket processor motherboard configuration ?, mostly perhaps researchers.

**Comment by graysky2** [comment]
@trippleflux  - Oddly enough, it build without error.  Regarding NUMA, I can drop the related symbols if it is problematic.

**Comment by graysky2** [comment]
Upon running that command a 2nd time, I was able to generate an error:
```
...
if cmp -s openvswitch-xen.spec.tmp xenserver/openvswitch-xen.spec; then touch xenserver/openvswitch-xen.spec; rm openvswitch-xen.spec.tmp; else mv openvswitch-xen.spec.tmp xenserver/openvswitch-xen.spec; fi
an.tmac:lib/ovs-fields.7:762: warning: tbl preprocessor failed, or it or soelim was not run; table(s) likely not rendered (TE macro called with TW register undefined)
troff:vswitchd/ovs-vswitchd.8:1290: warning: cannot select font 'CW'
make[5]: *** [Makefile:5684: manpage-check] Error 1
```

Complete [build log](https://gist.github.com/graysky2/25ca5e779f464827b78971f4242a7a33).

**Comment by trippleflux** [comment]
> Regarding NUMA, I can drop the related symbols if it is problematic. Are you certain there aren't gains to be had on single processor/multicore systems?

Sorry I am might be wrong, and I am not certain, no proof involved in my opinion, haven't tested it yet.
But we can be certain the following :
**_RHEL, FEDORA, DEBIAN or good more oldy/reliable linux distro is enabled it or becoming their [standards](https://hlandau.github.io/kconfigreport/option/CONFIG_NUMA_BALANCING.xhtml)._**

Regarding the following kernel config above from yoctoproject recommendations, I might be wrong, [I haven't know RT/preempt patch](https://wiki.linuxfoundation.org/realtime/start) is not by default applied on linux kernel main package.

Yocto  project [recommended ](https://lists.yoctoproject.org/g/linux-yocto/message/10737)the following IF we applied preempt patches : 
```
# CONFIG_NUMA_BALANCING is not set
# CONFIG_NUMA_BALANCING_DEFAULT_ENABLED is not set
# CONFIG TRANSPARENT_HUGEPAGE is not set
```
But as we are not applied preempt/RT patches, I think it's safe to use it all ?.

```
CONFIG_NUMA_BALANCING=y
CONFIG_NUMA_BALANCING_DEFAULT_ENABLED=y
CONFIG TRANSPARENT_HUGEPAGE=y
```

I think next steps for OpenWrt for x86_64 would be applying preempt/RT patches ?
```
https://cdn.kernel.org/pub/linux/kernel/projects/rt/5.15/
https://mirrors.edge.kernel.org/pub/linux/kernel/projects/rt/5.15/patches-5.15.119-rt65.tar.gz
```

A relevant 5.15.120 changes for "0003-arm64-mm-Make-arch_faults_on_old_pte-check-for-migra.patch" inside patches-5.15.119-rt65.tar.gz archive : 
```
diff -Naur a/arch/arm64/include/asm/pgtable.h b/arch/arm64/include/asm/pgtable.h
--- a/arch/arm64/include/asm/pgtable.h
+++ b/arch/arm64/include/asm/pgtable.h
@@ -983,6 +983,7 @@
 	 * us retaking a user fault which we just fixed up. The alternative
 	 * is doing a dsb(ishst), but that penalises the fastpath.
 	 */
+	 WARN_ON(is_migratable());
 }
 
 #define update_mmu_cache_pmd(vma, address, pmd) do { } while (0)
```

And
```
# CONFIG_NUMA_BALANCING is not set
# CONFIG_NUMA_BALANCING_DEFAULT_ENABLED is not set
# CONFIG TRANSPARENT_HUGEPAGE is not set
```


> warning: tbl preprocessor failed, or it or soelim was not run; table(s) likely not rendered (TE macro called with TW register undefined)
> troff:vswitchd/ovs-vswitchd.8:1290: warning: cannot select font 'CW'

For userspace? app like openvswitch, for me it's easier to ignore all warning with "**_TARGET_CFLAGS += -w_**" in the openvswitch Makefile.

**Comment by graysky2** [comment]
The preempt/RT patches are interesting ... maybe in a separate PR.  Removing the NUMA stuff allowed all checks to pass so probably no need to modify openvswitch, no?

**Comment by trippleflux** [comment]
> The preempt/RT patches are interesting ... maybe in a separate PR. Removing the NUMA stuff allowed all checks to pass so probably no need to modify openvswitch, no?

Yes I think so, submitting another PR for openvswitch would be not needed.

Feel free to submit The preempt/RT patches OpenWrt PR next if you wanted to.

Squeeze those performances out from OpenWrt for x86_64 & arm64/aarch64👍 .

**Comment by graysky2** [comment]
> Squeeze those performances out from OpenWrt for x86_64 & arm64/aarch64+1 .

Are any of the core packages written to be deterministic and thus benefit from real time kernel support?

**Comment by trippleflux** [comment]
@graysky2 
Could you elaborate more on your term about deterministic in OpenWrt ?

**Comment by graysky2** [comment]
Here is a good video @trippleflux - https://www.youtube.com/watch?v=sUDMG6ey9d0

I am asking if any of the core packages for routing or the like actually care about a RT kernel.  AFAIK, audio and video use-cases need RTK.  There are other use-cases beyond the scope of OpenWrt.

**Comment by trippleflux** [comment]
Well works already being started before on OpenWrt 
```
https://openwrt.org/docs/techref/xenomai
https://forum.openwrt.org/t/linux-rt-for-openwrt-is-calling-for-test/99997
https://source.denx.de/Xenomai/xenomai/-/wikis/home
```

From your own youtube reference, Afaik game engine and networking are latencies sensitive operations, even tough I haven't understand linux's internals. Perhaps experienced linux / OpenWrt developers care to join in?. 

Does preempt Dynamic or this real-time patch has really noticeable impact on  performance and reducing latency on vanilla version of user space daemons and applications?.

**Comment by trippleflux** [comment]
Just managed to finished compiled and fresh installed OpenWrt on x86_64 with those RT Patches : 
```
Linux Sentinel 5.15.120 #0 SMP PREEMPT_RT Sat Jul 15 06:50:35 2023 x86_64 GNU/Linux
```

Basic test on Cat 7 cable into my good old Intel i219-v NIC
test1


Normally my workstation waveform test result always nearly above +0-9 ms into my either simulated download or upload by waveform.

This is the craziest result, never ever encountered before with any router OS, especially our lovely OpenWrt!

test2

Above result (2nd screenshot is from my Vivo Android Smartphone connected into my my Xiaomi AX9000 running also OpenWrt which is connected into main upstream router running OpenWrt on x86_64, running waveform test on chrome browser.

My current conclusion, unless you wanted to run OpenWrt with all of them through good old cabling (Cat 5e, SFP+, etc), then we a doomed running RT patches especially if we have user with mostly using WIFI and double NAT into main router.

Seems RT patches is REALLY sensitive regarding client access latency. It will **_throw_** away your slow client in conclusion, especially your client's upstream bandwidth.

BUT overall OpenWrt Admin/Control Panel experiences is improved ALOT, other than those WIFI upload bandwidth big problems!.

**Comment by graysky2** [comment]
Nice result.  Can you share the commits you made in order to build the RT kernel?  I would like to see the same test you used on a OW system with the control being the stock OW kernel and the experiment being the RT kernel with all over variables held constant.  Maybe both runs on your hardware:

> Basic test on Cat 7 cable into my good old Intel i219-v NIC

**Comment by trippleflux** [comment]
Hello @graysky2 
  I haven't using git for my OpenWrt works, feel free to use this patch below : 

[RT_PATCHES_OpenWrt_Snapshot.zip](https://github.com/openwrt/openwrt/files/12075990/RT_PATCHES_OpenWrt_Snapshot.zip)

It should be a drop in patches for current OpenWrt snapshot, you might be able to apply it by put it at the root folder of your OpenWrt SRC dir and do "patch -p1 < RT_PATCHES_OpenWrt_Snapshot.patch".

You must not having any modifications / forks at all to apply the patches cleanly.

Have fun and I hope you will be also getting better finding than mine.

**Comment by graysky2** [comment]
> we should probably continue the rt discussion in the forums so as not to pollute the comments of this PR.

@trippleflux - Let's use [this](https://forum.openwrt.org/t/value-in-adding-preempt-rt-kernel-for-x86/165890) thread.

File: target/linux/x86/64/config-5.15 [hunk 1]
```diff
@@ -487,10 +491,11 @@ CONFIG_X86_INTEL_LPSS=y
 CONFIG_X86_INTEL_PSTATE=y
 CONFIG_X86_MINIMUM_CPU_FAMILY=64
 # CONFIG_X86_PCC_CPUFREQ is not set
-CONFIG_X86_PKG_TEMP_THERMAL=y
+# CONFIG_X86_PKG_TEMP_THERMAL is not set
```
Comment (hauke) [inline target/linux/x86/64/config-5.15:494]: Why do you deactivate `CONFIG_X86_PKG_TEMP_THERMAL` ?

File: target/linux/x86/64/config-5.15 [hunk 2]
```diff
@@ -264,11 +266,13 @@ CONFIG_INTEL_IOMMU_FLOPPY_WA=y
 # CONFIG_INTEL_IPS is not set
 # CONFIG_INTEL_MEI_HDCP is not set
 # CONFIG_INTEL_MENLOW is not set
-CONFIG_INTEL_PCH_THERMAL=y
+# CONFIG_INTEL_PCH_THERMAL is not set
```
Comment (hauke) [inline target/linux/x86/64/config-5.15:269]: Why do you deactivate `CONFIG_INTEL_PCH_THERMAL`?

File: target/linux/x86/64/config-5.15 (hunk 1 above)
Comment (graysky2) [reply target/linux/x86/64/config-5.15:494]: I understood `CONFIG_INTEL_HFI_THERMAL=y` to supply similar functionality making this setting redundant.  The model desktop distro I used to compare does not have this option enabled.  I can toggle it back if you feel it is needed.

File: target/linux/x86/64/config-5.15 (hunk 2 above)
Comment (graysky2) [reply target/linux/x86/64/config-5.15:269]: I understood `CONFIG_INTEL_HFI_THERMAL=y` to supply similar functionality making this setting redundant.  The model desktop distro I used to compare does not have this option enabled.  I can toggle it back if you feel it is needed.

**Comment by peterwillcn** [comment]
Add pci iommu and SR-IOV support

**Comment by hauke** [comment]
Please add support for kernel 6.12 too.

**Comment by graysky2** [comment]
@peterwillcn - what are the requisite ksymbols?
@hauke - 6.12?

**Comment by graysky2** [comment]
I am going to close this... I have a sloppy way of adding what I needed to my own build here: https://github.com/graysky2/openwrt/commit/8aa5d4002848a4624b2f951313bc13ac226b3fbe

I am not sure how to best distribute these throughout:
```
target/linux/x86/64/config-6.1
target/linux/x86/config-6.1
target/linux/x86/generic/config-6.1
```

**Comment by peterwillcn** [comment]
> @peterwillcn - what are the requisite ksymbols?

CONFIG_VFIO
CONFIG_VFIO_PCI
https://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html

**Pull request closed without merging**